patches/tModLoader/Terraria/GameContent/UI/EmoteBubble.cs.patch

```diff
@@ -71,6 +94,189 @@
+ 			}
+ 
++			// vector: draw position
++			// value2: bubble background frame
++			// flag: whether the emote bubble is actually showing
++			Rectangle emoteFrame = emote is >= 0 and < EmoteID.Count ? value.Frame(8, 39, emote * 2 % 8 + frame, 1 + emote / 4) : new Rectangle(0, 0, 34, 28);
++			
++			if (emote == -1) {
++				vector += new Vector2(effect.HasFlag(SpriteEffects.FlipHorizontally) ? 1 : (-1), -emoteFrame.Height + 3);
++			}
++			else {
++				if (ModEmoteBubble != null) {
++					value = ModContent.Request<Texture2D>(ModEmoteBubble.Texture).Value;
++					var moddedFrame = EmoteBubbleLoader.GetFrame(this);
++					emoteFrame = moddedFrame ?? value.Frame(2, 1, frame, 0);
++				}
++
++				if (!EmoteBubbleLoader.PreDraw(this, sb, value, vector, emoteFrame, origin, effect)) {
++					EmoteBubbleLoader.PostDraw(this, sb, value, vector, emoteFrame, origin, effect);
++					return;
++				}
++			}
++
+-			sb.Draw(value, vector, value2, Color.White, 0f, origin, 1f, effect, 0f);
++			sb.Draw(bubbleTexture, vector, value2, Color.White, 0f, origin, 1f, effect, 0f);
+ 			if (flag)
+ 				return;
+ 
+@@ -161,7 +_,7 @@
+ 					vector.X += 4f;
+ 				}
+ 
+-				sb.Draw(value, vector, value.Frame(8, 39, emote * 2 % 8 + frame, 1 + emote / 4), Color.White, 0f, origin, 1f, effect, 0f);
++				sb.Draw(value, vector, emoteFrame, Color.White, 0f, origin, 1f, effect, 0f);
+ 			}
+ 			else if (emote == -1) {
+ 				value = TextureAssets.NpcHead[metadata].Value;
+@@ -172,8 +_,10 @@
+ 				if ((float)value.Height / 16f > 1f / num)
+ 					num = 16f / (float)value.Height;
+ 
+-				sb.Draw(value, vector + new Vector2(effect.HasFlag(SpriteEffects.FlipHorizontally) ? 1 : (-1), -value2.Height + 3), null, Color.White, 0f, new Vector2(value.Width / 2, 0f), num, effect, 0f);
++				sb.Draw(value, vector, null, Color.White, 0f, new Vector2(value.Width / 2, 0f), num, effect, 0f);
+ 			}
++
++			EmoteBubbleLoader.PostDraw(this, sb, value, vector, emoteFrame, origin, effect);
+ 		}
+ 
+ 		private Vector2 GetPosition(out SpriteEffects effect) {
+@@ -208,6 +_,10 @@
+ 			}
+ 		}
+ 
++		/// <summary>
++		/// Send a emote from <see cref="Main.LocalPlayer"/>. Should never be called on server.
++		/// </summary>
++		/// <param name="emoteId"></param>
+ 		public static void MakeLocalPlayerEmote(int emoteId) {
+ 			if (Main.netMode == 0) {
+ 				NewBubble(emoteId, new WorldUIAnchor(Main.LocalPlayer), 360);
+@@ -217,8 +_,9 @@
+ 				NetMessage.SendData(120, -1, -1, null, Main.myPlayer, emoteId);
+ 			}
+ 		}
+-
+-		public void PickNPCEmote(WorldUIAnchor other = null) {
++		
++		public void PickNPCEmote(out bool useEmote, WorldUIAnchor other = null) {
++			useEmote = true;
+ 			Player plr = Main.player[Player.FindClosest(((NPC)anchor.entity).Center, 0, 0)];
+ 			List<int> list = new List<int>();
+ 			bool flag = false;
+@@ -261,8 +_,17 @@
+ 				ProbeCombat(list);
+ 			}
+ 
++			other ??= new WorldUIAnchor();
++			int modPickedEmote = NPCLoader.PickEmote((NPC)anchor.entity, plr, list, other);
```

Comment:
This should change to return `int?`
- `null` -> vanilla behavior
- `-1` -> no emote
- `int i` -> emote id `i`

Reply:
No need for the `out` variable then